### ISSUE TITLE
Connecting to unauthenticated Windows network shares (SMB/CIFS) needs asterisk in username field

### ISSUE THREAD

**Issue by shasheene** [rescuezilla/rescuezilla#190]
A user on the Rescuezilla Sourceforge forum has [wrote](https://sourceforge.net/p/rescuezilla/discussion/discussion/thread/a484b22dd2/#5c66) that he couldn't connect to an unauthenticated SMB/CIFS network share without settings the username to an asterisk character ("*"). The user also needed to set the version to "2.1" which is understandable for a Windows 7 based user:

> i used the ip address of the receiving computer and directory, populated the name field with an * asterisk and used 2.1 as version. it works like a charm. it seems the name and password fields that are labeled optional actually have to be populated. my system does not require a password but the name field must have an asterisk to work. hope this helps someone.

This issue was actually noted by [a GitHub user 8 months ago](https://github.com/rescuezilla/rescuezilla/issues/110#issuecomment-660688495) who ended up using an authenticated user, but I didn't look into it further at the time. The bug will have affected Rescuezilla v2.0-v2.1.3 (and also likely the versions prior to v2.0), and will have caused a bad user-experience for users.

The task is to ensure the username field is indeed optional, even for unauthenticated Windows network shared folder. [`man mount.cifs`](https://manpages.ubuntu.com/manpages/focal/man8/mount.cifs.8.html) says the following:

> username=arg|user=arg
>              specifies the username to connect as. If this is not given, then the environment variable USER is used.
>
> guest  don't prompt for a password.

The manpage above doesn't contain any information about how to connect to unauthenticated/anonymous shares, and says that a blank username leads to $USER being used if that argument is not provided. The issue is when the user doesn't provide a username, Rescuezilla does not populate any username argument, leading to $USER being used:

```
            if settings['username'] != "":
                credentials_string = "username=" + settings['username'] + "\n"

                            ...

            if credentials_string == "":
                smb_arguments += "guest"
            elif settings['password'] == "":
                    smb_arguments += ",guest"
```

**Therefore instead of specifying an asterisk, the best path appears for Rescuezilla to to set the mount.cifs' specify the username field as an empty string when that's what the user provides.** For example, on the command-line that would be: `mount.cifs '\\100.100.100.100\share' /mnt/backup -o username='',guest`. If that approach doesn't work, then providing an asterisk directly will have to be done.

Side note: it's unclear whether any similar issues apply to the SMB/CIFS password field or the domain field.

**If anybody has any input/ideas/suggestions on this, feel free to comment down below.**

**Comment by shasheene** [comment]
Ok so after further investigation it turned out the Windows anonymous SMB/CIFS network shares require the username field to be populated with _something_. There's nothing special about the asterisk character.

This issue has affected every single prior Rescuezilla release, but interestingly Linux-based Samba servers don't need the username field populated at all. I have confirmed that Linux-based Samba servers aren't negatively affected by specifying a non-blank username.

Therefore I have decided to always pass in the username "rescuezilla". This has the bonus effect of providing useful logging information to the system administrator of the SMB/CIFS server.
